bin/brew

```diff
@@ -40,6 +40,16 @@ fi
 
 HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX"
 
+# If we're running under macOS Rosetta 2, and it was requested by setting
+# HOMEBREW_CHANGE_ARCH_TO_ARM (for example in CI), then we re-exec this
+# same file under the native architecture
+if [[ "$HOMEBREW_CHANGE_ARCH_TO_ARM" = "1" ]] && \
+   [[ "$(uname -s)" = "Darwin" ]] && \
+   [[ "$(sysctl -n hw.optional.arm64 2>/dev/null)" = "1" ]] && \
+   [[ "$(sysctl -n sysctl.proc_translated 2>/dev/null)" = "1" ]]; then
+  exec arch -arm64e "$HOMEBREW_BREW_FILE" "$@"
+fi
```

Comment:
```suggestion
# If we're running under macOS Rosetta 2, and it was requested by setting
# HOMEBREW_CHANGE_ARCH_TO_ARM (for example in CI), then we re-exec this
# same file under the native architecture
if [[ "$HOMEBREW_CHANGE_ARCH_TO_ARM" = "1" ]] && \
   [[ "$(uname -s)" = "Darwin" ]] && \
   [[ "$(sysctl -n hw.optional.arm64 2>/dev/null)" = "1" ]] && \
   [[ "$(sysctl -n sysctl.proc_translated 2>/dev/null)" = "1" ]]; then
  exec arch -arm64e "$HOMEBREW_BREW_FILE" "$@"
fi
```

This logic looks good but can it go into `brew.sh` instead?

Reply:
I wanted it to happened as early as possible… we are calling brew many times during CI, so if we can re-exec as early as possible, I think it's better. Also, we need almost no information to do this.

If you prefer `brew.sh`, then I need help into where to put it. That thing is a beast, and I do not know where to put it.

Reply:
> If you prefer `brew.sh`, then I need help into where to put it. That thing is a beast, and I do not know where to put it.

At the very top seems fine to me 👍🏻

Reply:
PR updated